### ISSUE TITLE
A humble opinion :)

### ISSUE THREAD

**Issue by kuttan85** [reek/anti-adblock-killer#1930]
Respected sir,

I think we are approaching the limits of browser plugin based Ad-blockers were more and more websites started using Anti Adblocker techniques.  Manually bypassing each website's anti-adblocker is getting more difficult that calls for some radical upgrades needed to effectively bypass such sites with ease.  I recently tried Adguard for Windows **which is a standalone Adblocker program** which works fine on sites that uses Anti Adblocker techniques. I dont clearly know how Adguard for windows works but I suspect the program routes user internet traffic through their servers kind of VPN and is very effective.

Adguard now also offer DNS based adblocking service which is currently in Beta which I yet to try. All this points to the fact that website owners are more and more aware of Adblockers and they are constantly working to override it and web users badly needs a solution to Ads, Tracking and even forced malware download from sites if one doesn't use an Adblocker.

Many thanks to code gurus here that spend their time for millions of peoples benefit :+1:

**Comment by Giwayume** [comment]
Adguard does all the same exact things that uBlock or ABP does. There's no magical solution to stop triggering anti-adblock code globally.

Whether an advertisement resource is blocked by a client-side extension, VPN, or any other means, it's all the same to Javascript. It can still execute code when this occurs.

**Comment by X01X012013** [comment]
@Giwayume What about the a browser that tells JS resource loaded successfully when it's blocked by Adblocker?

**Comment by Giwayume** [comment]
@X01X012013 
It doesn't matter if you run a browser with hacked code that tries to arbitrarily identify advertisement HTML/JS and block it from executing. You still run into the same problem: advertisement implementations are **completely arbitrary** to the person writing the code. You have to handle it on a case by case basis, unless it's a shared solution like blockadblock.

No matter how you block the ads, you still have to make a decision at some point of what's an ad and what is not. What is anti adblocker code and what is not. The only way to solve that is to make a big list (which is what is done today) or make some sweeping assumptions about how ad code generally looks and end up accidentally blocking non-ads and essential features of a web page. No one wants to sacrifice a website not working altogether in order to bock a few annoying ads.

Also, no one wants to run a hacked browser. It will be buggy because it will have false positives, breaking web pages and frustrating the user, and it will have security holes which is one of the main driver behind why the web standards work the way they do today.

**Comment by X01X012013** [comment]
@Giwayume 
What I mean is the browser will work with Adblocker and only fake "success" statue for resources that Adblocker blocks. 
The browser can be based on Chromium, it's open source so no need to hack.

**Comment by Giwayume** [comment]
@X01X012013 
That still leads to the same issue, how do you identify which resources to give a false positive and which not to? It's a manual process no matter how you decide to do the blocking.

**Comment by X01X012013** [comment]
@Giwayume As I said, Adblocker will decide which resources to block, and all blocked resources will appear to be loaded. Yes the Adblocker will need a case by case "rule book" like how it works right now.

**Comment by rob641** [comment]
TL;DR: Adguard for window does significantly more than uBlock or ABP. 

There are actually quite a big difference between Adguard and other browser extensions. Someone above claiming that it does the same thing as browser extensions seems that they never actually tried Adguard.. But before beginning, I'd say Adguard does not route your traffic to a remote server as guessed in OP. Adguard DNS does, but standalone program never does it. 

Adguard support javascript rules. This is something like greasemonkey, but its way is more straightforward: it just injects the specified javascript inside a head node. 
This is why Adguard works well on sites using Anti-adblock. Some Anti-adblock techniques are tedious to nullify with ABP-style exception rules, and Adguard uses javascript rules for such sites. Those rules are contained in Adguard's own filter lists (which are publicly available).

Indeed, Adguard also supports userscript installation by extending their technology with injecting javascripts. You can even install Anti-adblock killer userscript to Adguard.


One more difference is that it supports page code filtering, which will never be available for browser extensions. What it does is a complete removal of a specified node in a html document, rather then hiding them as other adblockers does. This is possible because the standalone Adguard program filters pages before it reaches the browser by deep packet inspection. There is no api for browser extensions to do a similar thing.

You can use this arsenal to kill inline scripts used for anti-adblock. ublock blindly blocks all inline scripts, which often has undesirable effects when the website heavily uses inline js. Adguard allows to remove a specifically targeted script. This is particularly useful for end-users, because its syntax is pretty simple and finding out which script is used for anti-adblock is easy, for usually they are horribly obfuscated or contain words like "please disable adblock".

**Comment by reek** [comment]
Even if AdGuard is a more complete tool, there is not a single rule that can fight all the anti-adblock scripts, that is why we must create rules according to the mode of detection used.

Let us be magicians ✨

**Comment by kuttan85** [comment]
@Reek

Please try out AdGuard for Windows it is clearly more efficient in dealing with anti-adblock scripts than what browser plugin based Adblockers can do. The new AdGuard DNS based Adblocking too is more effective in bypassing anti-adblock scripts than browser plugin adblockers can.

**Comment by reek** [comment]
@kuttan85 
I can't test because it's a paid software, I have to buy a license

**Comment by uBlock-user** [comment]
> Adguard support javascript rules. This is why Adguard works well on sites using Anti-adblock.
This is something like greasemonkey, but its way is more straightforward: it just injects the specified javascript inside a head node. Some Anti-adblock techniques are tedious to nullify with ABP-style exception rules, and Adguard uses javascript rules for such sites. Those rules are contained in Adguard's own filter lists (which are publicly available).

What Adguard does is nothing spectacular. uBO already supports javascript injection, no big deal. Also AntiAdblock scripts are still triggered on Adguard untill they're manually patched up. FYI - You simply cannot compare Adguard for Windows with a browser extension like uBO or ABP.

There's no one defense solution for anti adblock scripts and never will be, as all of them are evolving to fight this `whack-a-mole` game. Whether you use Adguard or uBO or ABP, Anti Adblock scripts are still gonna pop up and you will have patch it up at least once until the webmasters give up.

**Comment by ghost** [comment]
@uBlock-user

Please tryout and compare both Adguard windows client Vs browser plugin based Adblockers. In terms of effectiveness in dealing with sites that uses Anti-Adblock scripts the Windows based Adguard software is clearly superior.  With Adguard windows those sites who regularly updates their anti-adblock scripts and problematic with browser based adblockers will display the site ads free without any issues as if the site doesn't used any anti-adblocker scripts.  

Recently I came to know Adguard introduced DNS based adblocker which when I tried was effective in dealing with sites that uses anti-adblock scripts. The only weakness of Adguard DNS adblocker is its slightly less adblocking capability but let you browse those sites without ads (mostly) who constantly updates their anti-adblocker scripts. In order for a totally ads free browsing experience i am using Adguard DNS and  uBlock Origin together where the Adguard DNS overrides sites anti-adblock scripts and ads and the remaining ads will be filtered out by the uBlock Origin plugin.

**Comment by uBlock-user** [comment]
> Please tryout and compare both Adguard windows client Vs browser plugin based Adblockers.

Windows based clients eat up your resources and need to be ran the minute you boot up your computer or else the protection won't work. Having said that, they act like anti-virus and run in without being sand-boxed which is bad since client you claim is an executable and I can't see the source. 

So I don't know what's running besides the client, could be a tracking code or what not. Basically. you're putting your machine at risk, your privacy at risk and your resources at a waste by running a Windows based client. 

> In terms of effectiveness in dealing with sites that uses Anti-Adblock scripts the Windows based Adguard software is clearly superior.

No proof, no validity, just plain display of your conjecture. uBO when ran in Hard Mode blocks everything including any and all scripts/css/images and what not which makes the website look broken, which can be mended but still it can take care in one swoop if you were to use such mode.  All you and that guy is doing is moaning about Adguard like a paid shill. uBO is far superior for a content blocker extension and there's nothing that uBO cannot block and the number of users are its proof. Adguard is only trying to copy which uBO established since the beginning.

>will display the site ads free without any issues as if the site doesn't used any anti-adblocker scripts.

That's nonsensical and simply won't work. If Anti Adblock is present, you will see it whether you're running uBO or Adguard, they cannot detect which extension are you running or a Windows based client, so it will display the warning regardless of what you use, keep your conjecture to yourself. This is the most illogical statement I have ever read.
